XRFoundation/XREngine
# Refactor Database Seeding

## Summary

- [x] refactors the feathers seeder to be entirely async and removes unused functionality
- [x] seeder must have 'FORCE_DB_REFRESH=true` explicitly set to run
- [x] change deployment flow to do check for unseeded database in builder rather than server pods
- [x] default project is no longer read-only on deployment, as all logical operations and tests are not dependent on it

Adds a couple new commands to the root package.json
`check-db-exists` which queries the database for if the 'users' table exists, if it does, it exits, if it does not, it runs the command
`init-db-production` which runs the API server with `APP_ENV=production` `FORCE_DB_REFRESH=true` which will seed the database and exit

## Checklist
- [x] Pre-push checks pass `npm run check`
  - [x] Linter passing via `npm run lint`
  - [x] Unit & Integration tests passing via `npm run test:packages`
  - [x] Docker build process passing via `npm run build-client`
- [ ] Changes reviewed by at least 2 approved reviewers

## QA Steps
```
docker container stop xrengine_db
docker container rm xrengine_db
docker container prune
npm run dev-docker
npm run check-db-exists
npm run install-projects
```
should destroy, recreate and seed database, then exit

## Reviewers

@barankyle
